embassy-executor/src/raw/mod.rs

```diff
@@ -581,14 +581,26 @@ impl embassy_time_queue_driver::TimerQueue for TimerQueue {
 #[cfg(feature = "integrated-timers")]
 embassy_time_queue_driver::timer_queue_impl!(static TIMER_QUEUE: TimerQueue = TimerQueue);
 
+#[cfg(all(feature = "rtos-trace", feature = "integrated-timers"))]
+const fn gcd(a: u64, b: u64) -> u64 {
+    if b == 0 {
+        a
+    } else {
+        gcd(b, a % b)
+    }
+}
+
 #[cfg(feature = "rtos-trace")]
 impl rtos_trace::RtosTraceOSCallbacks for Executor {
     fn task_list() {
         // We don't know what tasks exist, so we can't send them.
     }
     #[cfg(feature = "integrated-timers")]
     fn time() -> u64 {
-        Instant::now().as_micros()
+        const GCD_1K: u64 = gcd(embassy_time_driver::TICK_HZ, 1_000);
+        let ms_per_tick = (1000 / GCD_1K) / (embassy_time_driver::TICK_HZ / GCD_1K);
```

Comment:
original code uses microseconds, you're now doing milliseconds. [Docs](https://docs.rs/rtos-trace/latest/rtos_trace/trait.RtosTraceOSCallbacks.html#tymethod.time) say it should be microseconds.

Reply:
the order of operations here will accumulate error if tick rate is not a clean divisor of 1e6. Note:

- embassy_time: `(ticks * (1000 / GCD_1K)) / (embassy_time_driver::TICK_HZ / GCD_1K);`
- this PR: `ticks * ((1000 / GCD_1K) / (embassy_time_driver::TICK_HZ / GCD_1K));`

if you do the division first, it'll get rounded a bit, and that error will accumulate for every second passed. If you multiply first then divide, the rounding is just +/- 1us independently of the absolute value of the uptime.